File: 240331/총점 비교/compare-total-points.java
import java.util.Scanner;
import java.util.Arrays;
public class Main {
    public static class Student implements Comparable<Student>{
        String name;
        int kor;
        int eng;
        int math;
        public Student() {
            this.name = "";
            this.kor = 0;
            this.eng = 0;
            this.math = 0;
        }
        public Student(String name, int kor, int eng, int math) {
            this.name = name;
            this.kor = kor;
            this.eng = eng;
            this.math = math;
        }
        @Override
        public int compareTo(Student student) {
            return (this.kor +this.eng+this.math) - (student.kor + student.eng + student.math);
        }
    }
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Student[] students = new Student[n];
        for(int i=0;i<n;i++) {
            students[i] = new Student();
        }

        for(int i=0;i<n;i++) {
            students[i].name = sc.next();
            students[i].kor = sc.nextInt();
            students[i].eng = sc.nextInt();
            students[i].math = sc.nextInt();
        }
        Arrays.sort(students);
        for(int i=0;i<n;i++) {
            System.out.println(students[i].name + " " + students[i].kor + " " + students[i].eng + " " + students[i].math + " ");
        }
    }
}

File: 240418/총점 비교/compare-total-points.kt
import java.util.Scanner

class Student(
    val name: String,
    val kor: Int,
    val eng: Int,
    val math: Int
): Comparable<Student> {
    override fun compareTo(other: Student): Int {
        return (kor + eng + math) - (other.kor + other.eng + other.math)
    }
}

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    val students = List(n) {
        Student(sc.next(), sc.nextInt(), sc.nextInt(), sc.nextInt())
    }

    students.sorted().forEach {
        println("${it.name} ${it.kor} ${it.eng} ${it.math}")
    }
}